logic/function.lean

```diff
@@ -5,7 +5,7 @@ Authors: Johannes Hölzl, Mario Carneiro
 
 Miscellaneous function constructions and lemmas.
 -/
-import logic.basic data.option.defs
+import logic.basic data.option.defs data.set.function
```

Comment:
I'm against importing `data` (besides any `defs`) in `logic`

It's not officially documented, but my mental model is that `logic` should be mostly independent.

And then `order` -> `algebra`, and only small things are imported from `data`. We could move some stuff in `data.set...` to `data.set.defs`

Reply:
I did it the other way around, proving stuff in data.set.function